department-of-veterans-affairs/caseflow
# Craig/appeals 10138

Resolves [APPEALS-10138](https://vajira.max.gov/browse/APPEALS-10138)

### Description
- Added methods to models/vacols/correspondent.rb which will update a veteran's deceased date in VACOLS
- Added tests for all paths on the correspondent model and for the MPI controller which calls the methods
- Added a new seed file to let us create new specific test cases without changing the ID values for cases that may already be in use for testing

### Acceptance Criteria

### Testing Plan
1. Have an API Key generated in the rails console by a tech lead using [this wiki page](https://github.com/department-of-veterans-affairs/caseflow/wiki/Caseflow-APIs#api-keys)
2. In Postman, set the Authorization type to 'Bearer Token' and put the API key generated above into the 'Token' field
3. Set the method to POST and the url to {environment}/api/v1/mpi
4. Set the request body to the following, replacing the values for veterans_id and deceased_time as described : 
`{
    "veterans_id": "800000001",
    "deceased_ind": true,
    "deceased_time": "20111013"
}`
  - A return of `{ "success": 1 }` indicates that the record was updated, `{ "success": null }` indicates that it was not
5. Tech lead verify the veteran's SFNOD is updated in VACOLS using the rails console: `VACOLS::Correspondent.find_by(ssn: {veterans_id})` 
6. Change the veterans_id and deceased_time values to test at least the following scenarios:
  - Non-existent veterans_id value
  - Veteran who is already deceased, but has a different date of death than what is provided
  - Veteran who is already deceased and has the same date of death provided
  - "deceased_time" value of "null"
  - "deceased_ind" value of "null"
  - No Authorization token provided with request

Notes:
- Valid veterans_id values for this testing are 800000001, 800000002, 800000003, 800000004, 800000005
- Required deceased_time format is "YYYYMMDD"
